### PR TITLE
Enable reverse engineering of an exsiting Azure SQL Data Warehouse database

### LINKED ISSUE #15487: [REVENG] Unblock scaffolding of Azure SQL DW
See discussion here: https://github.com/aspnet/EntityFrameworkCore/pull/15438#issuecomment-486304475 
//cc @bricelam

### PR DESCRIPTION
**Pull request by ErikEJ**
fixes #15487

### CODE REVIEW

File: src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs [hunk 1]
```diff
@@ -536,12 +536,13 @@ private static string EscapeLiteral(string s)
...
     [c].[is_nullable],
     [c].[is_identity],
-    OBJECT_DEFINITION([c].[default_object_id]) AS [default_sql],
+    [df].[definition] AS [default_sql],
     [cc].[definition] AS [computed_sql]
 FROM [sys].[columns] AS [c]
 JOIN [sys].[tables] AS [t] ON [c].[object_id] = [t].[object_id]
 JOIN [sys].[types] AS [tp] ON [c].[user_type_id] = [tp].[user_type_id]
 LEFT JOIN [sys].[computed_columns] AS [cc] ON [c].[object_id] = [cc].[object_id] AND [c].[column_id] = [cc].[column_id]
+LEFT JOIN [sys].[default_constraints] AS [df] ON [c].[object_id] = [df].[parent_object_id] AND [df].[parent_column_id] = [c].[column_id]
```
Comment (smitpatel) [inline src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs]: nit: dc as alias rather than df.

Comment (smitpatel) [reply src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs]: another nit: align c before `=` makes it easier to transform to EF query in future if we do.

**Comment by ErikEJ** [comment]
@smitpatel Updated

**Comment by smitpatel** [comment]
@aspnet/build - to advice on the error in YAML file.

**Comment by dougbu** [comment]
@natemcmaster I think you were looking at the variable group issue? (BTW RaaS doesn't support this repo apparently: https://github.com/aspnet/AspNetCore-Internal/issues/2321)

**Comment by smitpatel** [comment]
@ErikEJ - Thanks for contribution.

I ran tests locally and they all passed fine.
Also pushing to master was failing so did merge from here
```
D:\code\ExternalContributions\EntityFrameworkCore [master ↑1]> git push
Counting objects: 7, done.
Delta compression using up to 6 threads.
Compressing objects: 100% (2/2), done.
Writing objects: 100% (7/7), 707 bytes | 707.00 KiB/s, done.
Total 7 (delta 5), reused 5 (delta 4)
remote: Resolving deltas: 100% (5/5), completed with 5 local objects.
remote: error: GH006: Protected branch update failed for refs/heads/master.
remote: error: Required status check "license/cla" is expected.
To github.com:aspnet/EntityFrameworkCore.git
 ! [remote rejected]     master -> master (protected branch hook declined)
error: failed to push some refs to 'git@github.com:aspnet/EntityFrameworkCore.git'
```
Any ideas @bricelam @natemcmaster ?

**Comment by natemcmaster** [comment]
Opened https://github.com/dotnet/core-eng/issues/6161. The error message you are seeing is consistent between PRs and CI builds.

**Pull request merged**